### PR TITLE
Making the path extension generated in xml, be optional

### PR DESCRIPTION
**Pull request by SauloSilva**
By default the `<loc>` is value the `file.html`.
Eg:

````xml
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd"
	xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
<url>

  <!-- the change influence this value -->
  <loc>foo.com/bar.html</loc>
  <!-- ------------------- -->

  <lastmod>2014-12-30T02:38:50.000Z</lastmod>
  <changefreq>daily</changefreq>
  <priority>0.5</priority>
</url>
</urlset>
````

I'm inserting a **extension attribute** that by default is true but if it is changed to false, the `.html` shall be replaced by `/`
Eg:
````xml
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd"
	xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
<url>

  <!-- value changed when extension is false -->
  <loc>foo.com/bar/</loc>
  <!-- ------------------- -->

  <lastmod>2014-12-30T02:38:50.000Z</lastmod>
  <changefreq>daily</changefreq>
  <priority>0.5</priority>
</url>
</urlset>
````

#### Implementation

````javascript
sitemap: {
      dist: {
          siteRoot: 'dist/',
          extension: false
      }
}
````

### CODE REVIEW

**Comment by thorsten** [comment]
+1, but the extension should be configurable as well, because e.g. I need http://www.example.org/foo instead of http://www.example.org/foo.html

**Comment by SauloSilva** [comment]
@thorsten exactly what I have done, to be configurable, you simply put `extension: false`, a look at the example I in the description of the pull request. Thank you.

**Comment by thorsten** [comment]
@SauloSilva yes, but you add an / at the end of the file as well, I forked your fork and removed the / for me

**Comment by SauloSilva** [comment]
@thorsten understand, I will do this also be configurable. Thank you.

**Comment by SauloSilva** [comment]
@benoitchantre and @thorsten I made the necessary improvements, just add `extension { required: false }` and if you like the trailing slash just add the attribute `trailingSlash: true`. I documented the two options [here](https://github.com/RayViljoen/grunt-sitemap/pull/25/files#diff-04c6e90faac2675aa89e2176d2eec7d8R89). 
Thanks for the suggested improvements guys.

**Comment by counterbeing** [comment]
Anyone have any complaints about this being merged into master?

File: README.md [hunk 1]
```diff
@@ -86,5 +86,31 @@ Set this to override `<changefreq>` in sitemap.
...
 Set this to override `<priority>` in sitemap.
 
+#### extension
+
+* Type: `object`
+* Default: `{ required: true }`
+
+Eg.
+
+````javascript
```
Comment (XhmikosR) [inline README.md]: `js` is shorter with the same effect.

File: README.md [hunk 2]
```diff
@@ -86,5 +86,31 @@ Set this to override `<changefreq>` in sitemap.
...
+extension: {
+  required: false
+}
+````
+
+and if you need to trailing slash simply add the attribute `trailingSlash: true` Eg.
```
Comment (XhmikosR) [inline README.md]: need to... missing verb?

File: README.md [hunk 3]
```diff
@@ -86,5 +86,31 @@ Set this to override `<changefreq>` in sitemap.
...
+
+````javascript
+extension: {
+  required: false
+  trailingSlash: true
+}
+````
+
+By default the `<loc>` is generated with a path that contains extension.
```
Comment (XhmikosR) [inline README.md]: "the extension"

File: README.md [hunk 4]
```diff
@@ -86,5 +86,31 @@ Set this to override `<changefreq>` in sitemap.
...
+Eg: '.hmlt' or '.htm'.
+To make your path without extension just add `extension: { required: false }` for this **required** of extension attribute.
```
Comment (XhmikosR) [inline README.md]: This needs rewording. CC @counterbeing

File: src/sitemap.coffee [hunk 5]
```diff
@@ -74,7 +76,18 @@ module.exports = (grunt) ->
 			if rawUrlPath.indexOf('/') is 0 then rawUrlPath = rawUrlPath.replace '/', ''
 
 			# Remove index.html
-			urlPath = rawUrlPath.replace /(index)\.[A-z]+$/, '', 'i'
+			rawUrlPath = rawUrlPath.replace /(index)\.[A-z]+$/, '', 'i'
+
+			urlPath = switch
+			    when typeof(extension) == 'object' && !extension.required && extension.trailingSlash
```
Comment (XhmikosR) [inline src/sitemap.coffee]: Are the brackets needed after `typeof` in coffeescript?

**Comment by XhmikosR** [comment]
@SauloSilva: please fecth and rebase against upstream/master. While you are it, squash the patches into one.

@counterbeing: made some comments. I'd wait before getting this merged.

**Comment by XhmikosR** [comment]
Also, since we don't have tests, someone should test this locally before merging.

File: src/sitemap.coffee (hunk 5 above)
Comment (counterbeing) [inline src/sitemap.coffee]: Not sure, but it makes immediate sense to me with them. I'd leave that personally.

**Comment by counterbeing** [comment]
I'm going to test it out right now. I think it's a good feature, I could imagine using it. I'm going to fix the readme typos and go from there.

File: src/sitemap.coffee (hunk 5 above)
Comment (XhmikosR) [inline src/sitemap.coffee]: As you can see in the generated Javascript, they are not needed. Not sure if they are in coffeescript though, but my guess would be they aren't.

**Comment by counterbeing** [comment]
Cool, thanks. I made a few typing corrections already, but if anything seems out of place, feel free to PR.

**Comment by thorsten** [comment]
I'm also using this already on www.phpmyfaq.de/sitemap.xml

**Comment by counterbeing** [comment]
Awesome possum, well, it's in the master now :)

**Comment by thorsten** [comment]
... and Google accepted the XML as well, the best test at all ;-)

**Pull request merged**